### PR TITLE
Add AUTH tests to Redis implementation

### PR DESCRIPTION
**Pull request by yfei1**
This commit added basic AUTH tests for Redis connections w/ and w/o sentinel. 

Depends on #1038.

The minisentinel library that we used for testing does not support AUTH from sentinel yet, so I manually set the value of `redis.sentinelUsePassword` to false when doing the end-to-end test. Added a TODO to change the test and enable the sentinel AUTH when minisentinel starts to supports it.

### CODE REVIEW

**Comment by sawagh** [comment]
Please merge and resolve so the PR is unblocked.

File: internal/statestore/redis.go [hunk 1]
```diff
@@ -71,10 +71,10 @@ func getHealthCheckPool(cfg config.View) *redis.Pool {
 
 	if cfg.IsSet("redis.sentinelHostname") {
 		sentinelAddr := getSentinelAddr(cfg)
-		healthCheckURL = redisURLFromAddr(sentinelAddr, cfg)
+		healthCheckURL = redisURLFromAddr(sentinelAddr, cfg, false)
```
Comment (sawagh) [inline internal/statestore/redis.go]: Why no password for the healthcheckurl? can you add a note about this to the PR description. The PR description just seems to state this being a test only change but looks like there is more.

Comment (yfei1) [reply internal/statestore/redis.go]: The redis sentinel library available in golang does not support authentication for now, so I only added the test case for sentinel w/o auth.

Comment (yfei1) [reply internal/statestore/redis.go]: working on an update

**Pull request merged**